### PR TITLE
Add all fragments except dashboard, to backstack

### LINKED ISSUE #682: Settings fragment should be in backstack
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    
### Bug Report    

When clicking back in settings section, the app shows toast to exit if back is pressed twice

**Expected Behaviour**

The settings fragment should be killed and app should return to the main screen on back press

### PR DESCRIPTION
**Pull request by mishuvs**
Fixes #682 

Changes: 
    1. All fragments are now being added to backstack after selection
    2. The background colour of SettingsFragment is changed to remove transparency (as it is added over DashboardFragment)

Screenshots of the change: 
backstack

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Shouldn't be transparent

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 1]
```diff
@@ -195,7 +210,9 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        ft.add(R.id.fragment_container, fragment).addToBackStack("from dashboard to fragment").commit();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: Extract the tag and use simpler string

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 2]
```diff
@@ -168,14 +176,21 @@ private void loadFragment(int navItemId) {
...
         Fragment fragment;
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+
         switch (navItemId) {
             case R.id.nav_dashboard:
                 fragment = EventDashboardFragment.newInstance(eventId);
-                break;
+                ft.replace(R.id.fragment_container, fragment).commit();
+                isDashboardActive = true;
+                return;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: Use break

Unconditionally set `isDashboardActive` to true before the switch

Outside of switch, check isDashBoardActive, and add or replace fragments accrodingly and commit together

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java (hunk 1 above)
Comment (mishuvs) [reply app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: It's just an optional name for the back state. So can I use `null`?

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 3]
```diff
@@ -172,6 +180,10 @@ private void loadFragment(int navItemId) {
         lastSelectedNavItemId = navItemId;
 
         Fragment fragment;
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+
+        isDashboardActive = true;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: Remove this

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 4]
```diff
@@ -195,7 +207,16 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        getSupportFragmentManager().popBackStack();
+        if (navItemId == R.id.nav_dashboard) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]:
```Java
isDashboardActive = navItemId == R.id.nav_dashboard;
if (isDashboardActive) {
```

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 5]
```diff
@@ -195,7 +207,16 @@ private void loadFragment(int navItemId) {
...
+            ft.replace(R.id.fragment_container, fragment);
+        } else {
+            ft.add(R.id.fragment_container, fragment).addToBackStack(null);
+            isDashboardActive = false;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: Remove

File: app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java [hunk 6]
```diff
@@ -195,7 +207,16 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        getSupportFragmentManager().popBackStack();
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: We want to pop it always?

Comment (mishuvs) [reply app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java]: We're adding to backstack in all cases except DashboardFragment. So I think, yes.

**Comment by iamareebjamal** [comment]
Selected fragment is not highlighted @mishuvs

**Pull request merged**